flink-core/src/main/java/org/apache/flink/api/common/io/OutputFormat.java

```diff
@@ -71,9 +71,10 @@
      * <p>When this method is called, the output format it guaranteed to be opened.
      *
      * @param record The records to add to the output.
-     * @throws IOException Thrown, if the records could not be added to to an I/O problem.
+     * @throws Exception Thrown, if the records could not be added due to an I/O problem or a
+     *     timeout.
      */
-    void writeRecord(IT record) throws IOException;
+    void writeRecord(IT record) throws Exception;
```

Comment:
-1; this change breaks source-compatibility with all existing implementations.

Need to double-check why japicmp didn't notice this.

Reply:
As said in the other comment, I **proposed** this change as `tryAcquire` already threw exceptions and `OutputFormat` is not user facing so I felt we could change to a more generic exception throwing. But it is true that even if it is not a user braking change, it is a source-compatibility breaking change. So if you prefer I'll make `tryAcquire` throw an unchecked exception in both Sink and OutputFormat so that we don't need to change the throw clause.

Reply:
> OutputFormat is not user facing

It is very much user-facing; they can (and do) implement their own formats and use it with existing sinks. So we must revert to throwing an unchecked exception.

Reply:
ok

Reply:
Do you want that I change the exception throwing to unchecked also on Sink for consistency with OutputFormat? If positive, the impact would be a breaking change for the user in exception management: in case of exception they will get RuntimeException(e) in place of e. I could also do the change only on the OutputFormat to avoid this breaking change in the sinks.

Reply:
I'll do the unchecked exception change only on `CassandraOutputFormat`  and not on `CassandraSink` to avoid a breaking change in exception management for sink users: avoid them getting `RuntimeException(e)` in place of `e` in case of exception.